dpbench/benchmarks/gpairs/gpairs_sycl_native_ext/gpairs_sycl/_gpairs_sycl.cpp

```diff
@@ -0,0 +1,44 @@
+//==- _gpairs_sycl.cpp - Python native extension of Black-Scholes   ===//
+//
+// Copyright 2022 Intel Corp.
+//
+// SPDX - License - Identifier : Apache 2.0
+///
+/// \file
+/// The files implements a SYCL-based Python native extension for the
+/// gpairs benchmark.
+
+#include "_gpairs_kernel.hpp"
+#include <dpctl4pybind11.hpp>
+
+void gpairs_sync(size_t nopt,
+                 size_t nbins,
+                 dpctl::tensor::usm_ndarray x1,
+                 dpctl::tensor::usm_ndarray y1,
+                 dpctl::tensor::usm_ndarray z1,
+                 dpctl::tensor::usm_ndarray w1,
+                 dpctl::tensor::usm_ndarray x2,
+                 dpctl::tensor::usm_ndarray y2,
+                 dpctl::tensor::usm_ndarray z2,
+                 dpctl::tensor::usm_ndarray w2,
+                 dpctl::tensor::usm_ndarray rbins,
+                 dpctl::tensor::usm_ndarray results)
+{
+    sycl::event res_ev = gpairs_impl(
+        x1.get_queue(), nopt, nbins, x1.get_data<float>(), y1.get_data<float>(),
```

Comment:
Why is the type `float`? 

We also need checks to ensure compatibility of the input arrays and a type check. Refer https://github.com/IntelPython/dpbench/blob/0fc5236865c4ea84849ea2bf3730964018bf0047/dpbench/benchmarks/black_scholes/black_scholes_sycl_native_ext/black_scholes_sycl/_black_scholes_sycl.cpp#L70